tests/Validator/LIVR/Rules/ListLengthTest.php

```diff
@@ -0,0 +1,109 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Validator\LIVR\Rules;
+
+use PHPUnit\Framework\TestCase;
+use uMotif\LIVR;
+
+final class ListLengthTest extends TestCase
+{
+    private const TEST_DATA_KEY = 'my_rule_list';
+    private const FORMAT_ERROR = 'FORMAT_ERROR';
+    private const REQUIRED = 'REQUIRED';
+
+    /**
+     * @dataProvider dataProvider
+     * @param $data
+     * @param array|null $ruleLength
+     * @param string|null $errorMessage
+     */
+    public function testItWorks($data, ?array $ruleLength, ?string $errorMessage = null): void
```

Comment:
This tests is checking for both positive and negative outcomes. I think it should be split so that we have a test that explicitly tests and asserts validation has passed and another that asserts validation has failed.